packages/cheqd-sdk/src/ledger/CheqdSdkLedgerService.ts

```diff
@@ -0,0 +1,107 @@
+import type { AbstractCheqdSDKModule, CheqdSDK, ICheqdSDKOptions, DidStdFee, DIDDocument } from '@cheqd/sdk'
+import type { SignInfo } from '@cheqd/ts-proto/cheqd/did/v2'
+import type { MsgCreateResourcePayload } from '@cheqd/ts-proto/cheqd/resource/v2'
+
+import { injectable } from '@aries-framework/core'
+import { createCheqdSDK, DIDModule, ResourceModule, CheqdNetwork } from '@cheqd/sdk'
+import { DirectSecp256k1HdWallet, DirectSecp256k1Wallet } from '@cosmjs/proto-signing'
+import { EnglishMnemonic as _ } from '@cosmjs/crypto'
+import { fromString } from 'uint8arrays'
+
+import { CheqdSdkModuleConfig } from '../CheqdSdkModuleConfig'
+
+export interface CheqdSdkLedgerConfig {
+  networkType?: CheqdNetwork
+  rpcUrl?: string
+  cosmosPayerSeed?: string
+}
+
+export enum DefaultRPCUrl {
+  Mainnet = 'https://rpc.cheqd.net',
+  Testnet = 'https://rpc.cheqd.network',
+}
+
+export class EnglishMnemonic extends _ {
+  public static readonly _mnemonicMatcher = /^[a-z]+( [a-z]+)*$/
+}
+
+@injectable()
+export class CheqdSdkLedgerService {
+  private readonly cosmosPayerWallet: Promise<DirectSecp256k1HdWallet | DirectSecp256k1Wallet>
```

Comment:
Would it be possible to dynamically set the payer wallet for each transaction? We've just removed the global `publicDidSeed` from the agent for indy-sdk and now require you to provide a `submitterDid` when writing to the indy ledger, just so it's more flexible in what the options are. 

Maybe you control multiple dids, or for example in multi-tenancy mode this could switch based on the agent context.

For indy we require you to add the key for the submitter to the wallet first, and then reference that in the input parameters, after which the wallet signs the request.

Reply:
Not sure how this is possible, it will lead to the race conditions again you mentioned in this [comment](https://github.com/hyperledger/aries-framework-javascript/pull/1334#discussion_r1114200987)

A cosmos account is not binded to a DID/wallet like publicDIDSeed of indy-sdk

If it's just to manage different identity wallets owned by the same entity, there is one cosmos account, which is used just to pay the network fees in all the wallets. There can be two separate account's mainnet/testnet.

If it's entirely a different entity, then it's better for them to use a different agent rather than using multitenancy. As the mnemonic/seed of the cosmos account is exposed in the config unless we have a config for each multitenant wallet.

Reply:
thanks for the info. I'm less familiar with how the cheqd sdk works. But couldn't we use one sdk instantiation per network. They then each get a `CheqdPool` or just manage them in the CheqdSdkLedgerService, and that doesn't have a payer key and we sign the transactions with the wallet?

Ideally we sign everything with the wallet, as that makes it more aligned with how we manage keys in AFJ (they never leave the wallet).

If our wallet would support Secpk256K would that be possible?

Reply:
yea I'm taking this up. It's better to have one instantiation.

Will discuss with the team on your second point, I agree it's better to manage the keys within AFJ.